base/regex.jl

```diff
@@ -15,6 +15,8 @@ type Regex
     extra::Ptr{Void}
     ovec::Vector{Csize_t}
     match_data::Ptr{Void}
+    capture_name_to_idx::Dict{Symbol, Int}
+    idx_to_capture_name::Dict{Int, Symbol}
```

Comment:
+1. This is not going to be something you want to use for truly high-performance code anyway – for that you'll want to do the indexed lookup.